src/cuga/backend/cuga_graph/policy/tool_guard/tool_guard_runtime.py

```
@@ -0,0 +1,607 @@
+            await self._runtime.guard_toolcall(
+                tool_name=function_name,
+                args=arguments | {"args": args_obj},
+                delegate=self.invoker,
+            )
+        except PolicyViolationException as e:
+            error = str(e)
+            logger.warning(
+                f"Tool guard blocked call to '{function_name}': {error}"
+            )
+            return error
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)